### PR TITLE
Show version banner in docs

### PR DESCRIPTION
**Pull request by jezdez**
### Description

This follows [the docs](https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/version-dropdown.html#version-dropdowns), not sure if this also works in the scenario the conda docs are under, since there is [some auto-detection happening](https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/readthedocs.html) around the version switcher.

### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -98,6 +98,15 @@
     "navbar_start": ["navbar-logo"],
     "use_edit_page_button": True,
     "goatcounter_url": "https://docs-conda-io.goatcounter.com/count",
+    "show_version_warning_banner": True,
+    "switcher": {
+        # This should live in "latest" (=in-development version) since
+        # we can modify the supported versions in the version switcher
+        # "json_url": "https://docs.conda.io/projects/conda/en/latest/_static/switcher.json",
```
Comment (jezdez) [inline docs/source/conf.py:105]:
```suggestion
        # "json_url": "https://docs.conda.io/projects/conda/en/latest/_static/switcher.json",
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -98,6 +98,15 @@
...
+        "json_url": "https://continuumio-conda-docs--13693.com.readthedocs.build/projects/conda/en/13693/_static/switcher.json",
```
Comment (jezdez) [inline docs/source/conf.py:106]:
```suggestion
        "json_url": "https://docs.conda.io/projects/conda/en/latest/_static/switcher.json",
```

**Review by jezdez — COMMENTED** [review]
We can't really test this until we've merged the switcher.json file into main, where it would be served via RTD.

**Review by travishathaway — APPROVED** [review]
Yes, this is tricky to test in our staging environment. I'm okay with testing it live.

**Pull request merged**